src/Riok.Mapperly/Helpers/EnumMappingBuilderHelper.cs

```diff
@@ -0,0 +1,86 @@
+using Microsoft.CodeAnalysis;
+using Riok.Mapperly.Abstractions;
+using Riok.Mapperly.Descriptors;
+using Riok.Mapperly.Diagnostics;
+
+namespace Riok.Mapperly.Helpers;
+
+public static class EnumMappingBuilderHelper
+{
+    public static HashSet<IFieldSymbol> GetMembers(this ITypeSymbol symbol, HashSet<IFieldSymbol> ignoredMembers) =>
+        symbol.GetMembers().OfType<IFieldSymbol>().Where(x => !ignoredMembers.Remove(x)).ToHashSet(SymbolTypeEqualityComparer.FieldDefault);
+
+    public static void AddUnmatchedSourceIgnoredMembers(this MappingBuilderContext ctx, ISet<IFieldSymbol> ignoredUnmatchedSourceMembers)
```

Comment:
For the member mapping builder a `MemberMappingDiagnosticReporter` class is used. I'd follow the same approach here, maybe create a `EnumMappingDiagnosticReporter` class, align the method names and move the methods.

Reply:
Yes, agreed. I have created an EnumMappingDiagnosticReporter class and moved the methods as suggested.